src/cpp/subscriber/StatisticsParticipantListener.cpp

```diff
@@ -59,7 +82,7 @@ void StatisticsParticipantListener::process_endpoint_discovery(
 
     // Get the participant from the database
     GUID_t endpoint_guid = info.info.guid();
-    GUID_t participant_guid(endpoint_guid.guidPrefix, EntityId_t());
+    GUID_t participant_guid(endpoint_guid.guidPrefix, c_EntityId_RTPSParticipant);
```

Comment:
I am missing the header file that declares this constant: `fastdds/rtps/common/EntityId_t.hpp`